File: Java/724. Find Pivot Index.java
class Solution {
    public int pivotIndex(int[] nums) { 
        int sum = 0;
        int left = 0;
        for (int i : nums)
        sum+=i;
        for (int i = 0; i<nums.length; i++){
            sum -= nums[i];
//            System.out.printf("%d,%d,%d\n",i,sum,left);
            if(sum == left)
            return i;
            left += nums[i];
        }
        return -1;
    }
}

File: Python/724_Find_Pivot_Index.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        summ = sum(nums)
        left = 0
        for i, x in enumerate(nums):
            if left == summ-left-x:
                return i
            left += x
        return -1